commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/twod/hull/ConvexHull2D.java

```diff
@@ -0,0 +1,469 @@
+        this.vertices = Collections.unmodifiableList(new ArrayList<>(vertices));
+        this.path = buildHullPath(vertices, precision);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<Vector2D> getVertices() {
+        return vertices;
+    }
+
+    /** Get a path defining the convex hull. The path will contain
+     * <ul>
+     *      <li>zero segments if the hull consists of only a single point,</li>
+     *      <li>one segment if the hull consists of two points,</li>
+     *      <li>three or more segments defining a closed loop if the hull consists of more than
+     *          two non-collinear points.</li>
+     * </ul>
+     * @return polyline path defining the convex hull
+     */
+    public LinePath getPath() {
+        return path;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConvexArea getRegion() {
+        return path.isClosed() ?
+                ConvexArea.convexPolygonFromPath(path) :
+                null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder();
+        sb.append(getClass().getSimpleName())
+            .append("[vertices= ")
+            .append(getVertices())
+            .append(']');
+
+        return sb.toString();
+    }
+
+    /** Build a polyline representing the path for a convex hull.
+     * @param vertices convex hull vertices
+     * @param precision precision context used to compare floating point values
+     * @return path for the convex hull defined by the given vertices
+     */
+    private static LinePath buildHullPath(final Collection<Vector2D> vertices,
+            final Precision.DoubleEquivalence precision) {
+        if (vertices.size() < 2) {
+            return LinePath.empty();
+        }
+
+        final boolean closeLoop = vertices.size() > 2;
+
+        return LinePath.builder(precision)
+                .appendVertices(vertices)
+                .build(closeLoop);
+    }
+
+    /** Class used to build convex hulls. The builder is based on the Akl-Toussaint
+     * heuristic to construct the hull. The heuristic is based on the idea of a
+     * convex quadrilateral, which is formed by four points with the lowest and
+     * highest x / y coordinates. Any point that lies inside this quadrilateral can
+     * not be part of the convex hull and can thus be safely discarded before
+     * generating the convex hull itself.
+     * <p>
+     * The complexity of the operation is O(n), and may greatly improve the time it
+     * takes to construct the convex hull afterwards, depending on the point
+     * distribution.
+     *
+     * @see <a href=
+     *      "http://en.wikipedia.org/wiki/Convex_hull_algorithms#Akl-Toussaint_heuristic">
+     *      Akl-Toussaint heuristic (Wikipedia)</a>
+     */
+    public static final class Builder {
+
+        /** Corner of triangle with minimal x coordinate. */
```

Comment:
Q. `triangle` to `quadrilateral`?

Reply:
Yes, that should obviously mean quadrilateral.